cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -212,6 +224,71 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	return err
 }
 
+func (s *Service) getAddresses(ctx context.Context, vm compute.VirtualMachine) ([]corev1.NodeAddress, error) {
+
+	addresses := []corev1.NodeAddress{}
+
+	if vm.NetworkProfile.NetworkInterfaces != nil {
+		for _, nicRef := range *vm.NetworkProfile.NetworkInterfaces {
+
+			// The full ID includes the name at the very end. Split the string and pull the last element
+			// Ex: /subscriptions/$SUB/resourceGroups/$RG/providers/Microsoft.Network/networkInterfaces/$NICNAME
+			// We'll check to see if ID is nil and bail early if we don't have it
+			if nicRef.ID == nil {
+				continue
+			}
+			explodedNicID := strings.Split(to.String(nicRef.ID), "/")
+			nicName := explodedNicID[len(explodedNicID)-1]
+
+			// Fetch nic and append its addresses
+			nicInterface, err := networkinterfaces.NewService(s.Scope).Get(ctx, &networkinterfaces.Spec{Name: nicName})
+			if err != nil {
+				return addresses, err
+			}
+			nic, ok := nicInterface.(network.Interface)
+			if !ok {
+				return addresses, errors.New("unexpected type for network interface")
+			}
+
+			if nic.IPConfigurations != nil {
+				for _, ipConfig := range *nic.IPConfigurations {
+					if ipConfig.PrivateIPAddress != nil {
+						addresses = append(addresses,
+							corev1.NodeAddress{
+								Type:    corev1.NodeInternalIP,
+								Address: to.String(ipConfig.PrivateIPAddress),
+							},
+						)
+					}
+
+					if ipConfig.PublicIPAddress != nil {
```

Comment:
simplify the branch 

```
if ipConfig.PublicIPAddress == nil {
    continue
}
```

also can you please move it to separate function

getPublicIPAddress(ipConfig.PublicIPAddress.ID)